src/main/java/net/minecraftforge/common/extensions/IForgeJukebox.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Parrot;
+import net.minecraft.world.entity.animal.allay.Allay;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.JukeboxBlock;
+import net.minecraft.world.level.block.entity.JukeboxBlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+
+import javax.annotation.Nullable;
+
+/**
+ * Extension added to deal with mojang hardcoding the logic for playing records and parrot/allay dancing to only the vanilla jukebox.
+ */
+public interface IForgeJukebox
+{
+
+    /**
+     * @param entity The entity inserting into jukebox. This can be null.
+     * @param level The level of the jukebox.
+     * @param pos The position of the jukebox.
+     * @param state The blockstate of the jukebox
+     * @param stack The itemstack being inserted.
+     */
+    default void insertRecord(@Nullable Entity entity, LevelAccessor level, BlockPos pos, BlockState state, ItemStack stack) {
+        if (this instanceof JukeboxBlock jukebox) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    default void insertRecord(@Nullable Entity entity, LevelAccessor level, BlockPos pos, BlockState state, ItemStack stack)
    {
        if (this instanceof JukeboxBlock jukebox)
        {
```